base/math.jl

```diff
@@ -1239,7 +1239,7 @@ julia> rem2pi(7pi/4, RoundDown)
 """
 function rem2pi end
 function rem2pi(x::Float64, ::RoundingMode{:Nearest})
-    isnan(x) && return NaN
+    (x-x !== 0.0) && return NaN
```

Comment:
```suggestion
    !isfinite(x) && return NaN
```

Reply:
At the moment `isfinite(::AbstractFloat)` is https://github.com/JuliaLang/julia/blob/783a6aa4d3e5ad371a5cec5d7510f46d423042e3/base/float.jl#L497 but perhaps it should be `x - x !== zero(x)`, if that's more efficient?  Or even `iszero(x - x)`, and `iszero` should be `x === zero(x)`